src/dataframe/dataframe.jl

```diff
@@ -368,7 +380,7 @@ end
 function Base.setindex!(df::DataFrame,
                 v::Any,
                 col_ind::ColumnIndex)
-    insert_single_column!(df, upgrade_scalar(df, v), col_ind)
+    fill_single_column!(df, v, col_ind)
```

Comment:
Good point. Unfortunately `df[:col_ind] .= v` does not work (even on v0.6) due to `broadcast!` being a little particular with what data types it considers "vectors", such as `String`. Also, I recall there was a discussion about `setindex!(A, v, Colon())` still being slower than `fill!(A, v)`, though I don't know if this issue has been addressed by now.